### PR TITLE
Swap from PF Grid component to custom grid to better fit widget content.

### PR DESCRIPTION
**Pull request by dvail**
## Description

~Updates the layout of the dashboard to use a custom css grid (thanks @alanonthegit) instead of the built in PatternFly Grid component. The PF Grid component relies on responsive breakpoints and classes applied via React props in order to determine the size of columns and how many columns a widget should span. This isn't a great fit for the dashboard, as a fixed column span at specific breakpoints either did not make optimal use of space or cause widget headers to overflow.~

Updates the dashboard grid to use the `<Gallery>` component, which is almost identical to the original custom CSS.

## Checklist
- [ ] Investigated and inspected CI test results

## Testing Performed

Test view at default 13" Macbook resolution (~1500px)
image

Shrink the browser ensuring no titles overflow. The grid will convert to single column when the browser is ~1400px wide.
image
image

Dropping down to ~1200px, the PatternFly sidebar will no longer take up horizontal space and will be set to overlay on the content behind it. This gives the widgets room to switch back to two columns until about ~1090px.
image
image

At around 540px, the single column layout will have a horizontal overflow instead of breaking widget layout.
image

Very large screens will adjust to a 3 column layout.
image

### CODE REVIEW

**Comment by dvail** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2138** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2138" target="_blank">Graphite</a> 
    * **PR #2166** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2166" target="_blank">Graphite</a> 
    * **PR #2190** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2190" target="_blank">Graphite</a> 
    * **PR #2204** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2204" target="_blank">Graphite</a> 
      * **PR #2238** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2238" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/2238?utm_source=stack-comment).

File: ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/ComplianceLevelsByStandard.tsx [hunk 1]
```diff
@@ -129,7 +129,7 @@ function ComplianceLevelsByStandard() {
             header={
                 <Flex direction={{ default: 'row' }}>
                     <FlexItem grow={{ default: 'grow' }}>
-                        <Title headingLevel="h2">Compliance coverage by standard</Title>
+                        <Title headingLevel="h2">Compliance by standard</Title>
```
Comment (dvail) [inline ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/ComplianceLevelsByStandard.tsx:132]: This buys us 70px on the minimum widget size, which translates into 140px of extra space before we drop to a single column layout.

File: ui/apps/platform/src/Containers/Dashboard/PatternFly/DashboardPage.css [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.widget-grid {
+  --min-widget-width: 510px;
```
Comment (dvail) [inline ui/apps/platform/src/Containers/Dashboard/PatternFly/DashboardPage.css:2]: This should prevent headers from overflowing on widgets until the "# Active aging images widget" exceeds 10,000 images total in the title.

**Review by vjwilson — APPROVED** [review]
I'm not a fan of going off-script for styles, so I think that after this ships, we should present this use case to the PatternFly team, so that they can adopt it somehow as part of their offering.

World domination, one CSS class at a time.

**Comment by alanonthegit** [comment]
@dvail this looks great! Thanks for your hard work. I'm exploring a way to condense the options button, which will let us shrink cards a bit more and make the headers more future-proof. That should translate to minor tweaks to the min-width and max-width values. Will follow up with a mock.

**Comment by dvail** [comment]
@vjwilson I took a closer look at this, and PatternFly is way ahead of us. Customizing the <Grid> component was what I was looking at originally, but they offer a <Gallery> component which is essentially this exact use case. World domination will have to wait.

@alanonthegit The only thing I didn't include (to avoid custom CSS) was a max-width on the grid, so someone with an enormous screen will see 4, 5, or more widgets side-by-side. Is this OK or do we still want to enforce a limit here?

**Comment by alanonthegit** [comment]
@dvail Oh great! I still think we should enforce a max limit of 3 per row or else the page will become unbalanced.

**Pull request merged**